library/alloc/src/boxed.rs

```diff
@@ -1161,14 +1161,19 @@ impl<T: ?Sized, A: Allocator> Box<T, A> {
     /// This conversion does not allocate on the heap and happens in place.
     ///
     /// This is also available via [`From`].
-    #[unstable(feature = "box_into_pin", issue = "62370")]
+    ///
+    /// # Notes
+    ///
+    /// Because `Box` and `Pin` are `#[fundamental]`, implementing `From<Box>`
+    /// for `Pin<T>` may be ambiguous.
```

Comment:
Fundamental is an unstable (and not very well known) feature, so this seems like an odd thing to document on such a potentially prominent impl/type without links or further elaboration.

I think the impl text is also wrong, it should be `From<Box<T>> for Pin<T>`, right?

Is this an impl that's possible to add outside std? Maybe if the `T` is concrete? The blanket impl definitely seems like it would be alloc-only.

Reply:
I think we should say something more concrete here, like "it's not recommended that crates add such an impl, as it'll introduce ambiguity, and give an example, rather than talking about fundamental -- that would make more sense to me.

Reply:
Updated the docs via 41beb30, how about?